test/replay/btcstaking_test.go

```diff
@@ -40,75 +39,307 @@ func FuzzCreatingAndActivatingDelegations(f *testing.F) {
 	f.Fuzz(func(t *testing.T, seed int64) {
 		t.Parallel()
 		r := rand.New(rand.NewSource(seed))
-		numFinalityProviders := uint32(datagen.RandomInRange(r, 3, 7))
-		numDelegationsPerFinalityProvider := uint32(datagen.RandomInRange(r, 20, 30))
-
+		numFinalityProviders := datagen.RandomInRange(r, 2, 3)
+		numDelegationsPerFinalityProvider := datagen.RandomInRange(r, 1, 2)
 		driverTempDir := t.TempDir()
 		replayerTempDir := t.TempDir()
-		driver := NewBabylonAppDriver(t, driverTempDir, replayerTempDir)
+		driver := NewBabylonAppDriver(r, t, driverTempDir, replayerTempDir)
 		// first finalize at least one block
-		driver.GenerateNewBlock(t)
-		stakingParams := driver.GetBTCStakingParams(t)
+		driver.GenerateNewBlock()
 
-		fpInfos := GenerateNFinalityProviders(r, t, numFinalityProviders, driver.GetDriverAccountAddress())
+		scenario := NewStandardScenario(driver)
+		scenario.InitScenario(numFinalityProviders, numDelegationsPerFinalityProvider)
 
-		// register all finality providers
-		for _, fpInfo := range fpInfos {
-			driver.SendTxWithMsgsFromDriverAccount(t, fpInfo.MsgCreateFinalityProvider)
-		}
+		// Replay all the blocks from driver and check appHash
+		replayer := NewBlockReplayer(t, replayerTempDir)
+		replayer.ReplayBlocks(t, driver.GetFinalizedBlocks())
+		// after replay we should have the same apphash
+		require.Equal(t, driver.GetLastState().LastBlockHeight, replayer.LastState.LastBlockHeight)
+		require.Equal(t, driver.GetLastState().AppHash, replayer.LastState.AppHash)
+	})
+}
 
-		// register all delegations
-		var allDelegationInfos []*datagen.CreateDelegationInfo
-		for _, fpInfo := range fpInfos {
-			delInfos := GenerateNBTCDelegationsForFinalityProvider(
-				r,
-				t,
-				numDelegationsPerFinalityProvider,
-				driver.GetDriverAccountAddress(),
-				fpInfo,
-				stakingParams,
-			)
-			allDelegationInfos = append(allDelegationInfos, delInfos...)
-			msgs := ToCreateBTCDelegationMsgs(delInfos)
-			driver.SendTxWithMsgsFromDriverAccount(t, msgs...)
-		}
+func TestNewAccountCreation(t *testing.T) {
+	r := rand.New(rand.NewSource(time.Now().UnixNano()))
+	driverTempDir := t.TempDir()
+	replayerTempDir := t.TempDir()
+	driver := NewBabylonAppDriver(r, t, driverTempDir, replayerTempDir)
+	// first finalize at least one block
+	driver.GenerateNewBlock()
+
+	stakers := driver.CreateNStakerAccounts(5)
```

Comment:
We should pass the random down to account creation as well, it could be in a future improvement